tests/reader/reader_test.go

```diff
@@ -10,70 +12,165 @@ import (
 	"github.com/supitsdu/clipper/tests"
 )
 
-func TestReadAll_Files(t *testing.T) {
-	content1 := "file1 content"
-	content2 := "file2 content"
-
-	file1, err := tests.CreateTempFile(t, content1)
-	if err != nil {
-		t.Fatalf("Failed to create temporary file: %s", err)
-	}
-	file2, err := tests.CreateTempFile(t, content2)
-	if err != nil {
-		t.Fatalf("Failed to create temporary file: %s", err)
-	}
-
-	config := &options.Config{
-		FilePaths: []string{file1.Name(), file2.Name()},
-	}
-	contentReader := reader.ContentReader{Config: config}
+func TestReadAll(t *testing.T) {
+	t.Run("reads single file", func(t *testing.T) {
+		file1, err := tests.CreateTempFile(t, tests.SampleText32)
+		if err != nil {
+			t.Fatalf("Failed to create temporary file: %s", err)
+		}
 
-	result, err := contentReader.ReadAll()
-	assert.NoError(t, err)
-	expectedResult := content1 + "\n" + content2 + "\n"
-	assert.Equal(t, expectedResult, result)
-}
+		config := &options.Config{
+			FilePaths: []string{file1.Name()},
+		}
 
-func TestReadFile(t *testing.T) {
-	content := "file content"
-	file, err := tests.CreateTempFile(t, content)
-	if err != nil {
-		t.Fatalf("Failed to create temporary file: %s", err)
-	}
+		contentReader := reader.ContentReader{Config: config}
 
-	config := &options.Config{}
-	contentReader := reader.ContentReader{Config: config}
+		result, err := contentReader.ReadAll()
+		assert.NoError(t, err)
+		expectedResult := tests.SampleText32 + "\n"
+		assert.Equal(t, expectedResult, result)
+	})
+
+	t.Run("reads from argument", func(t *testing.T) {
+		config := &options.Config{
+			Text: tests.SampleText32,
+		}
+
+		contentReader := reader.ContentReader{Config: config}
 
-	result, err := contentReader.ReadFile(file.Name())
-	assert.NoError(t, err)
-	assert.Equal(t, content, result)
+		result, err := contentReader.ReadAll()
+		assert.NoError(t, err)
+		expectedResult := tests.SampleText32 + "\n"
+		assert.Equal(t, expectedResult, result)
+	})
+
+	t.Run("reads multiple files", func(t *testing.T) {
+		fileContents := []string{"Japan", "Australia", "Germany"}
+		expectedResults := strings.Join(fileContents, "\n") + "\n"
+		var filePaths []string
+
+		for index := 0; index < 3; index++ {
+			file, err := tests.CreateTempFile(t, fileContents[index])
+			if err != nil {
+				t.Fatalf("Failed to create temporary file: %s", err)
+			}
```

Comment:
```suggestion
			require.NoErrorf("Failed to create temporary file: %s", err)
			}
```

You already have testify, you should use it

Reply:
In fact, this could be improved

You created an helper to create a temp file

You are testing the err outside the createtempfile, you should add the check on err in the helper with require.NoError. 

This way you will only return the file, and not the error